include/pybind11/pybind11.h

```diff
@@ -1328,20 +1328,30 @@ template <typename... Args> struct init_alias {
 
 
 inline void keep_alive_impl(handle nurse, handle patient) {
-    /* Clever approach based on weak references taken from Boost.Python */
     if (!nurse || !patient)
         pybind11_fail("Could not activate keep_alive!");
 
     if (patient.is_none() || nurse.is_none())
         return; /* Nothing to keep alive or nothing to be kept alive by */
 
-    cpp_function disable_lifesupport(
-        [patient](handle weakref) { patient.dec_ref(); weakref.dec_ref(); });
+    auto tinfo = get_type_info(Py_TYPE(nurse.ptr()));
```

Comment:
This should be: `auto tinfo = all_type_info(Py_TYPE(nurse.ptr())); if (!tinfo.empty()) { ... }` so that it can support a nurse which is an instance of a Python class inheriting from multiple pybind-registered bases.